src/server/rest/v1/service/ChargingStationService.ts

```diff
@@ -114,12 +116,40 @@ export default class ChargingStationService {
     if (Utils.objectHasProperty(filteredRequest, 'forceInactive')) {
       chargingStation.forceInactive = filteredRequest.forceInactive;
     }
+    if (Utils.objectHasProperty(filteredRequest, 'manualConfiguration')) {
+      // Check manual config
+      if (!chargingStation.manualConfiguration && filteredRequest.manualConfiguration) {
+        chargingStation.manualConfiguration = filteredRequest.manualConfiguration;
+        delete chargingStation.templateHash;
+        delete chargingStation.templateHashCapabilities;
+        delete chargingStation.templateHashOcppStandard;
+        delete chargingStation.templateHashOcppVendor;
+        delete chargingStation.templateHashTechnical;
+      } else if ((chargingStation.manualConfiguration && !filteredRequest.manualConfiguration) ||
+      (!filteredRequest.manualConfiguration && !(chargingStation.chargePoints?.length > 0))) {
+        // If charging station is not configured manually anymore, the template will be applied again
+        chargingStation.manualConfiguration = filteredRequest.manualConfiguration;
+        const templateUpdateResult = await OCPPUtils.enrichChargingStationWithTemplate(req.user.tenantID, chargingStation);
+        // If not template was found, throw error (Check is done on technical configuration)
+        if (!templateUpdateResult.technicalUpdated) {
+          throw new AppError({
+            source: Constants.CENTRAL_SERVER,
+            action: action,
+            errorCode: HTTPError.GENERAL_ERROR,
+            message: `Error occurred while updating chargingStation: '${chargingStation.id}'. No template found`,
+            module: MODULE_NAME, method: 'handleUpdateChargingStationParams',
+            user: req.user,
+          });
+        }
+        rebootRequired = true;
+      }
+    }
     // Existing Connectors
     if (!Utils.isEmptyArray(filteredRequest.connectors)) {
       for (const filteredConnector of filteredRequest.connectors) {
         const connector = Utils.getConnectorFromID(chargingStation, filteredConnector.connectorId);
         // Update Connectors only if no Charge Point is defined
-        if (connector && Utils.isEmptyArray(chargingStation.chargePoints)) {
+        if (connector && (Utils.isEmptyArray(chargingStation.chargePoints) || chargingStation.manualConfiguration)) {
```

Comment:
I do not really know what is meant here. 
We have to check for the manual configuration, because when there are charge points and the manual configuration is activated, we have to apply the manually configured connector properties.

Reply:
You shall not consider no chargePoints = manualConfiguration here. 
You can have chargePoints on manually configured stations and then not override the chargePoints.

Reply:
Sorry I do not understand it.

> You shall not consider no chargePoints = manualConfiguration.
But in fact if there are no charge points, the user needs to configure the connectors himself or am I wrong here? 

> You can have chargePoints on manually configured stations and then not override the chargePoints.
If we do not overwrite the charge points on a manual configured station, we are not able to configure them.

Reply:
> > You shall not consider no chargePoints = manualConfiguration.
> > >
> 
> But in fact if there are no charge points, the user needs to configure the connectors himself or am I wrong here?

When you switch a charging station from template to manual, you reuse UI side all data for that charging stations, including chargPoints (hidden, maybe not fully user configurable UI side but fully there). And the dashboard is going to push the charging station data, including chargePoints.  

> 
> > You can have chargePoints on manually configured stations and then not override the chargePoints.
> > >
> 
> If we do not overwrite the charge points on a manual configured station, we are not able to configure them.

Sure but you do not need extra checks, the current code properly handle the chargePoints data wherever it comes: template or manual, we do not care. The code just use what is given in input with the existing sanity checks.